### PR TITLE
Upgrade netty to 4.1.75

### LINKED ISSUE #15142: Upgrade netty to 4.1.75
### Description of the problem / feature request:

Please upgrade Netty 4.1.69 -> 4.1.75

On linux, we frequently run into https://github.com/netty/netty/issues/11815

This is likely solved in Netty 4.1.75 via https://github.com/netty/netty/commit/1df38d2c7be98e11fa13c7ee52aadbfbea7669e6

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

There is no easy repro, it seems to be a race condition.  Only seems to appear on Linux, not Mac.  See below for stack trace.

### What operating system are you running Bazel on?

CircleCI image:

```
Build-agent version 1.0.117438-72c2fbdd (2022-03-29T17:30:21+0000)
System information:
 Server Version: 20.10.12
 Storage Driver: overlay2
  Backing Filesystem: xfs
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Kernel Version: 5.13.0-1017-aws
 Operating System: Ubuntu 20.04.4 LTS
 OSType: linux
 Architecture: x86_64
 ```

### What's the output of `bazel info release`?

```
root@62d8983e8522:/src/mn/projects/fullstory# bazel version
Bazelisk version: v1.4.0
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
Build label: 5.0.0
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Jan 19 14:08:54 2022 (1642601334)
Build timestamp: 1642601334
Build timestamp as int: 1642601334


root@62d8983e8522:/src/mn/projects/fullstory# bazel info release
INFO: Invocation ID: 92faf5b1-4236-49ef-a704-9c2ee74d5bcb
INFO: Reading 'startup' options from /src/mn/projects/fullstory/.bazelrc.user: --shutdown_on_low_sys_mem, --max_idle_secs=5
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=0 --terminal_columns=179
INFO: Reading rc options for 'info' from /src/mn/projects/fullstory/.bazelrc:
  Inherited 'common' options: --enable_platform_specific_config --repository_cache=./bazel-cache/repos --experimental_repository_cache_hardlinks
INFO: Reading rc options for 'info' from /src/mn/projects/fullstory/.bazelrc.user:
  Inherited 'common' options: --repository_cache=/tmp/bazel/repo_cache
INFO: Reading rc options for 'info' from /src/mn/projects/fullstory/.bazelrc:
  Inherited 'build' options: --strategy=TypeScriptCompile=worker --remote_cache=https://bazel-cache.ops.fs.team --google_default_credentials --remote_upload_local_results=false --disk_cache=./bazel-cache --incompatible_remote_results_ignore_disk --incompatible_exclusive_test_sandboxed --define=GCR_REPOSITORY=fs-ops/feature-builds --define=GCR_REGISTRY_PREFIX=gcr.io --incompatible_strict_action_env --workspace_status_command=./tools/bazelbuild/load_build_info.sh
INFO: Reading rc options for 'info' from /src/mn/projects/fullstory/.bazelrc.user:
  Inherited 'build' options: --show_progress --build_tag_filters=-no-ci --local_cpu_resources=2 --local_ram_resources=2048 --action_env=NODE_OPTIONS --worker_max_instances=TypeScriptCompile=1 --discard_analysis_cache --nokeep_state_after_build --notrack_incremental_state --announce_rc --google_credentials=/root/gcloud-service-key.json --disk_cache=/tmp/bazel/action_cache --remote_upload_local_results=true --jobs 10
release 5.0.0
```

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

```
fatal: not a git repository (or any of the parent directories): .git
```

###  Have you found anything relevant by searching the web?

Yes.. see above Netty issue. https://github.com/netty/netty/issues/11815

### Any other information, logs, or outputs that you want to share?

```
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ActionLookupData{actionLookupKey=ConfiguredTargetKey{label=//go/src/fs/services/alexa/main/alexa:alexa, config=BuildConfigurationValue.Key[48a7a349feeea004bb328fa5c50a7b593c6e17033e218e2151db12a21b868035]}, actionIndex=1}' (requested by nodes 'TargetCompletionKey{topLevelArtifactContext=com.google.devtools.build.lib.analysis.TopLevelArtifactContext@90904c3b, actionLookupKey=ConfiguredTargetKey{label=//go/src/fs/services/alexa/main/alexa:alexa, config=BuildConfigurationValue.Key[48a7a349feeea004bb328fa5c50a7b593c6e17033e218e2151db12a21b868035]}, willTest=false}', 'ActionLookupData{actionLookupKey=ConfiguredTargetKey{label=//go/src/fs/services/alexa/main/alexa:alexa, config=BuildConfigurationValue.Key[48a7a349feeea004bb328fa5c50a7b593c6e17033e218e2151db12a21b868035]}, actionIndex=4}')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:674)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:382)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: io.netty.handler.codec.DecoderException: javax.net.ssl.SSLException: error:1e000065:Cipher functions:OPENSSL_internal:BAD_DECRYPT
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:477)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
Caused by: javax.net.ssl.SSLException: error:1e000065:Cipher functions:OPENSSL_internal:BAD_DECRYPT
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.shutdownWithError(ReferenceCountedOpenSslEngine.java:1071)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.sslReadErrorResult(ReferenceCountedOpenSslEngine.java:1365)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1305)
	at io.netty.handler.ssl.ReferenceCountedOpenSslEngine.unwrap(ReferenceCountedOpenSslEngine.java:1392)
	at io.netty.handler.ssl.SslHandler$SslEngineType$1.unwrap(SslHandler.java:217)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1342)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:507)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:446)
	... 21 more
```

### PR DESCRIPTION
**Pull request by vpanta**
In order to fix https://github.com/netty/netty/issues/11815 for users of `bazel`, this upgrades the libraries under `third_party/netty` to `netty/netty@4.1.75`, along with `third_party/netty_tcnative` to `netty/netty_tcnative@2.0.51`.

This change needs to also include a `grpc-java` update to `1.45.1`, to fix https://github.com/grpc/grpc-java/issues/8981 which had been preventing `netty/netty@4.1.75` from working properly.

This PR has 3 parts/commits, which if necessary can be split and merged separately:
1. Split out a `third_party/grpc-java` from `third_party/grpc`
   - This makes a 1:1 mapping of GithHub repo to `third_party` directory.
   - Also makes it clearer in practice for these to be different versions (notable for commit 2).
2. Upgrade to `grpc/grpc-java@1.45.1`
   - All jars came from https://repo1.maven.org/maven2/io/grpc/
   - I had attempted to upgrade to `grpc/grpc@1.45.0` as well, however I ran into [issues in the build that were beyond my understanding to fix](https://github.com/bazelbuild/bazel/pull/15145#issuecomment-1084090100). However, the attempt of updating `grpc-java` alone was successful.
3. Upgrade to `netty/netty@4.1.75` and `netty/netty_tcnative@2.0.51`
   - All jars came from https://repo1.maven.org/maven2/io/netty/
   - This change also pushed Bazel's size to just over 311MB, so changed the integration test appropriately.

### CODE REVIEW

**Comment by vpanta** [comment]
Not as simple as I'd hoped, will look deeper tomorrow.

**Comment by vpanta** [comment]
This isn't ready for review unfortunately, I'm still trying to get `gRPC` updated in addition to `netty` (due to https://github.com/grpc/grpc-java/issues/8981), but now I'm getting more incompatibilities. Currently, I either get stuck behind one of the following two errors:
```
ERROR: /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/com_github_grpc_grpc/BUILD:5264:23: error loading package '@com_google_googleapis//google/rpc': Unable to find package for @com_google_googleapis_imports//:imports.bzl: The repository '@com_google_googleapis_imports' could not be resolved: Repository '@com_google_googleapis_imports' is not defined. and referenced by '@com_github_grpc_grpc//:google_rpc_status_upb'
```
which is resolvable by removing some of the patched-out contents, but the next blocker is:
```
ERROR: /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/com_github_grpc_grpc/BUILD:5254:23: error loading package '@com_github_cncf_udpa//xds/data/orca/v3': at /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/com_github_cncf_udpa/bazel/api_build_system.bzl:1:6: at /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/com_envoyproxy_protoc_gen_validate/bazel/pgv_proto_library.bzl:1:6: at /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/io_bazel_rules_go/proto/compiler.bzl:25:5: at /private/var/tmp/_bazel_vasilios/a66b1d3a8126aa917c2980f966604cd1/external/io_bazel_rules_go/go/private/rules/transition.bzl:32:5: Unable to find package for @io_bazel_rules_go_name_hack//:def.bzl: The repository '@io_bazel_rules_go_name_hack' could not be resolved: Repository '@io_bazel_rules_go_name_hack' is not defined. and referenced by '@com_github_grpc_grpc//:xds_orca_upb'
```
which is its own bag of "I don't know how to untangle this"...

**Comment by vpanta** [comment]
Ok, this is now ready for review. It's three distinct commits, each of which is probably viewed/reviewed on their own. I've updated the PR request above to go more into details about each commit and their purpose.

**Review by coeuvre — APPROVED** [review]
Thanks!

Can you rebase on `master`?

**Comment by coeuvre** [comment]
And unfortunately, we cannot import a PR that contains both `third_party` and non `third_party` changes. Can you split the PR into multiple small PRs that each of them works and only contains only `third_party` change or non `third_party` change?

**Comment by coeuvre** [comment]
I understand that moving `grpc-java` to `third_party/grpc-java` make it clearer but may introduce troubles when splitting PRs. I recommend first focus on only upgrading gRPC in one PR and clean it up in another one but it's up to you.

**Comment by vpanta** [comment]
No worries, sounds good, I wasn't sure the right organization of the PRs, but I can rebase/re-structure it for sure. I'll revert this to a Draft for now, and re-open the PRs when they're ready.

**Review by coeuvre — APPROVED** [review]
@meteorcloudy Can you help import this?

File: third_party/BUILD [hunk 1]
```diff
@@ -588,12 +589,12 @@ UNNECESSARY_DYNAMIC_LIBRARIES = select({
 genrule(
     name = "filter_netty_dynamic_libs",
     srcs = select({
-        "//src/conditions:darwin_arm64": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final-osx-aarch_64.jar"],
-        "//src/conditions:darwin_x86_64": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final-osx-x86_64.jar"],
-        "//src/conditions:linux_aarch64": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final-linux-aarch_64.jar"],
-        "//src/conditions:linux_x86_64": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final-linux-x86_64.jar"],
-        "//src/conditions:windows": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final-windows-x86_64.jar"],
-        "//conditions:default": ["netty_tcnative/netty-tcnative-boringssl-static-2.0.44.Final.jar"],
```
Comment (3405691582) [inline third_party/BUILD:596]: This leaves the default case broken. Should this just be `netty-tcnative-classes-2.0.51.Final.jar`, or is something more elaborate required?

**Pull request closed without merging**